reaper.go

```diff
@@ -36,69 +37,75 @@ var (
 // The ContainerProvider interface should usually satisfy this as well, so it is pluggable
 type ReaperProvider interface {
 	RunContainer(ctx context.Context, req ContainerRequest) (Container, error)
-	Config() TestContainersConfig
+	Config() *properties.TestcontainersConfig
 }
 
 // NewReaper creates a Reaper with a sessionID to identify containers and a provider to use
 func NewReaper(ctx context.Context, sessionID string, provider ReaperProvider, reaperImageName string) (*Reaper, error) {
-	mutex.Lock()
-	defer mutex.Unlock()
-	// If reaper already exists re-use it
-	if reaper != nil {
-		return reaper, nil
-	}
+	if reaper == nil {
+		mutex.Lock()
+		defer mutex.Unlock()
+
+		if reaper == nil {
+			dockerHost := extractDockerHost(ctx)
+
+			listeningPort := nat.Port("8080/tcp")
+
+			req := ContainerRequest{
+				Image:        reaperImage(reaperImageName),
+				ExposedPorts: []string{string(listeningPort)},
+				NetworkMode:  Bridge,
+				Labels: map[string]string{
+					TestcontainerLabel:         "true",
+					TestcontainerLabelIsReaper: "true",
+				},
+				SkipReaper: true,
```

Comment:
should we avoid using it due to it is deprecated?

Reply:
Indeed. I was pushing this PR first and then remove all usages of SkipReaper from the code. I'm OK adding that to this PR